lib/cargoDeny.nix

```diff
@@ -0,0 +1,26 @@
+{ cargo-deny
+, lib
+, mkCargoDerivation
+}:
+
+{ cargoDenyExtraArgs ? ""
+, cargoExtraArgs ? ""
+, src
+, ...
+}@origArgs:
+let
+  args = builtins.removeAttrs origArgs [
+    "cargoDenyExtraArgs"
+    "cargoExtraArgs"
+  ];
+in
+mkCargoDerivation (args // {
+  buildPhaseCargoCommand = ''
+    cargo --offline ${cargoExtraArgs} \
+      deny ${cargoDenyExtraArgs} check bans licenses sources
```

Comment:
Maybe we should make `bans licenses sources` the default value of `cargoDenyExtraArgs` so that consumers can easily change it if they want to test for something else